ihmc-common-walking-control-modules/src/main/java/us/ihmc/commonWalkingControlModules/momentumBasedController/optimization/DynamicsMatrixCalculator.java

```diff
@@ -2,99 +2,149 @@
+           toolbox.getMassMatrixCalculator(),
+           toolbox.getGravityCoriolisExternalWrenchMatrixCalculator(),
+           toolbox.getContactWrenchMatrixCalculator(),
+           toolbox.getRhoSize());
+   }
 
-      helper = new DynamicsMatrixCalculatorHelper(coriolisMatrixCalculator, jointIndexHandler);
-      helper.setRhoSize(rhoSize);
+   public DynamicsMatrixCalculator(FloatingJointBasics rootJoint,
+                                   JointIndexHandler jointIndexHandler,
+                                   CompositeRigidBodyMassMatrixCalculator massMatrixCalculator,
+                                   GravityCoriolisExternalWrenchMatrixCalculator coriolisMatrixCalculator,
+                                   ContactWrenchMatrixCalculator contactWrenchMatrixCalculator,
+                                   int rhoSize)
+   {
+      this.massMatrixCalculator = massMatrixCalculator;
+      this.coriolisMatrixCalculator = coriolisMatrixCalculator;
+      this.contactWrenchMatrixCalculator = contactWrenchMatrixCalculator;
+      this.rhoSize = rhoSize;
 
-      int numberOfDoFs = jointIndexHandler.getNumberOfDoFs();
-      int floatingBaseDoFs = rootJoint != null ? rootJoint.getDegreesOfFreedom() : 0;
-      bodyDoFs = numberOfDoFs - floatingBaseDoFs;
+      degreesOfFreedom = jointIndexHandler.getNumberOfDoFs();
+      floatingBaseDoFs = rootJoint != null ? rootJoint.getDegreesOfFreedom() : 0;
+      bodyDoFs = degreesOfFreedom - floatingBaseDoFs;
 
       jointTorques = new DMatrixRMaj(bodyDoFs, 1);
 
-      coriolisMatrix = new DMatrixRMaj(numberOfDoFs, 1);
-      contactForceJacobian = new DMatrixRMaj(rhoSize, numberOfDoFs);
+      massMatrix = new DMatrixRMaj(degreesOfFreedom, degreesOfFreedom);
+      coriolisAndGravityMatrix = new DMatrixRMaj(degreesOfFreedom, 1);
+      contactForceJacobian = new DMatrixRMaj(rhoSize, degreesOfFreedom);
 
-      floatingBaseMassMatrix = new DMatrixRMaj(floatingBaseDoFs, numberOfDoFs);
-      floatingBaseCoriolisMatrix = new DMatrixRMaj(floatingBaseDoFs, 1);
+      floatingBaseMassMatrix = new DMatrixRMaj(floatingBaseDoFs, degreesOfFreedom);
+      floatingBaseCoriolisAndGravityMatrix = new DMatrixRMaj(floatingBaseDoFs, 1);
       floatingBaseContactForceJacobian = new DMatrixRMaj(rhoSize, floatingBaseDoFs);
 
-      bodyMassMatrix = new DMatrixRMaj(bodyDoFs, numberOfDoFs);
-      bodyCoriolisMatrix = new DMatrixRMaj(bodyDoFs, 1);
+      bodyMassMatrix = new DMatrixRMaj(bodyDoFs, degreesOfFreedom);
+      bodyCoriolisAndGravityMatrix = new DMatrixRMaj(bodyDoFs, 1);
       bodyContactForceJacobian = new DMatrixRMaj(rhoSize, bodyDoFs);
       bodyContactForceJacobianTranspose = new DMatrixRMaj(bodyDoFs, rhoSize);
 
       torqueMinimizationObjective = new DMatrixRMaj(bodyDoFs, 1);
    }
 
-   public void reset()
+   public void setGravity(double gravityZ)
    {
-      coriolisMatrixCalculator.setExternalWrenchesToZero();
+      coriolisMatrixCalculator.setGravitionalAcceleration(-Math.abs(gravityZ));
```

Comment:
Dunno how much this suggestion would ripple over the code, but imo it's a bit sketch to have a negate in here. I see elsewhere we have `gravityZ = 9.81` which is then negated here. I'd personally much rather us have `gravityZ=  -9.81` up top which is just passed straight through here. I'll point out where I see it.

Reply:
Thought as much -- happy to disregard this then